daemon/start.go

```diff
@@ -36,9 +36,6 @@ func (daemon *Daemon) ContainerStart(name string, hostConfig *runconfig.HostConf
 				return err
 			}
 			container.Unlock()
-			if err := daemon.adaptContainerSettings(hostConfig, false); err != nil {
```

Comment:
@ibuildthecloud This code path is for !windows, that doesn't matter, anyway I won't concern that because:
1. This path is for backword compatibility, which is deprecated for a very long time, so the usage is rare.
2. `adaptContainerSettings` in this path was added just very short ago, the impact is very limited.
3. The main change here is we call `adaptContainerSettings` later, adapt after verify is what we do for create, that should not be a problem.

What do you mean about "InitDNSHostConfig needed some change" and what issue would that cause?